src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -0,0 +1,59 @@
+import { RepositoryError } from "@domain/errors"
+import { OnChainError, MakeTxFilter, MakeTxDecoder } from "@domain/bitcoin/onchain"
+import { MakeWalletsRepository } from "@services/mongoose/wallets"
+import { MakeLedgerService } from "@services/ledger"
+import { MakeOnChainService } from "@services/lnd/onchain-service"
+import { toLiabilitiesAccountId, LedgerError } from "@domain/ledger"
+import { LOOK_BACK } from "@core/utils"
+import { ONCHAIN_MIN_CONFIRMATIONS } from "@config/app"
+import { WalletTransactionHistory } from "@domain/wallets"
+
+// TODO should be exposed via PriceSerivce / LiquidityProvider
+import { getCurrentPrice } from "@services/realtime-price"
+
+export const getTransactionsForWalletId = async ({
+  walletId,
+}: {
+  walletId: WalletId
+}) => {
+  const wallets = MakeWalletsRepository()
+  const wallet = await wallets.findById(walletId)
+  if (wallet instanceof RepositoryError) return wallet
+  return getTransactionsForWallet(wallet)
+}
+
+export const getTransactionsForWallet = async (
+  wallet: Wallet,
+): Promise<WalletTransaction[] | CoreError> => {
+  const ledger = MakeLedgerService()
+  const liabilitiesAccountId = toLiabilitiesAccountId(wallet.id)
+  const ledgerTransactions = await ledger.liabilityTransactions(liabilitiesAccountId)
+  if (ledgerTransactions instanceof LedgerError) return ledgerTransactions
+
+  const confirmedHistory = WalletTransactionHistory.fromLedger(ledgerTransactions)
+
+  const onChain = MakeOnChainService(MakeTxDecoder(process.env.NETWORK as BtcNetwork))
+  if (onChain instanceof OnChainError) {
+    return confirmedHistory.transactions
+  }
+
+  const onChainTxs = await onChain.getIncomingTransactions(LOOK_BACK)
+  if (onChainTxs instanceof OnChainError) {
+    return confirmedHistory.transactions
+  }
+
+  const filter = MakeTxFilter({
+    confsLT: ONCHAIN_MIN_CONFIRMATIONS,
+    addresses: wallet.onChainAddresses,
+  })
+  const pendingTxs = filter.apply(onChainTxs)
+
+  // TODO should be a service - not a function call
+  let price = await getCurrentPrice()
+  if (typeof price !== "number") {
+    price = 0
```

Comment:
I think we may never want `price = 0` should be a fallback value. for multiple reasons:
- if we need to do a conversion, we may need to divide by this value. this would lead to further issues
- it could be some form of an attack vector eventually for USD account

Reply:
so I think at this point we either want to have `NaN` or trigger an error.

Reply:
maybe that doesn't matter here because it's only for pending transaction